torch/csrc/autograd/autograd_meta.cpp

```diff
@@ -88,7 +88,8 @@ bool has_same_meta(const Variable& base, const Variable& other) {
   if (!base.defined() || !other.defined()) {
     return false;
   }
-  if (base.storage_offset() != other.storage_offset()) {
+  if (base.storage_offset() != other.storage_offset() &&
+      !(base.numel() == 0 && other.numel() == 0)) {
```

Comment:
```suggestion
if (base.numel() == 0 && other.numel() == 0) {
  return true; //0-element tensors are always fine
}
  if (base.storage_offset() != other.storage_offset()  {
```

Reply:
Good catch. I also updated it so that the storage numel and conj check are reordered above.